From bd989a24fe287ea443c6f9e8478f86e79fcab001 Mon Sep 17 00:00:00 2001
From: Robert Lehmann <robert.lehmann@sitec-systems.de>
Date: Thu, 10 Mar 2016 13:27:51 +0100
Subject: [PATCH] S Core: Initial commit of specific score dts

Initial commit of device specific score dts. This dts file was splitted
from the main kernel repository.

Ticket #1023
---
 arch/arm/boot/dts/imx6sx-score.dts | 195 +++++++++++++++++++++++++++++++++++++
 1 file changed, 195 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx6sx-score.dts

diff --git a/arch/arm/boot/dts/imx6sx-score.dts b/arch/arm/boot/dts/imx6sx-score.dts
new file mode 100644
index 0000000..9325bc7
--- /dev/null
+++ b/arch/arm/boot/dts/imx6sx-score.dts
@@ -0,0 +1,195 @@
+/*
+ * Copyright (C) 2014-2015 Freescale Semiconductor, Inc.
+ * Copyright (C) 2015, 2016 Robert Lehmann <robert.lehmann@sitec-systems.de>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+/dts-v1/;
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+#include "imx6sx-score.dtsi"
+
+/ {
+    model = "sitec systems S Core Eval-Board";
+    compatible = "fsl,imx6sx-score", "fsl,imx6sx";
+
+    chosen {
+        stdout-path = &uart1;
+    };
+
+    memory {
+        reg = <0x80000000 0x20000000>;
+    };
+
+    leds {
+        compatible = "gpio-leds";
+        alive {
+            label = "Alive";
+            gpios = <&gpio3 19 GPIO_ACTIVE_HIGH>;
+            linux,default-trigger = "heartbeat";
+        };
+        led {
+            label = "Red_led";
+            gpios = <&gpio3 20 GPIO_ACTIVE_HIGH>;
+        };
+    };
+
+};
+
+&cpu0 {
+    operating-points = <
+        /* kHz    uV */
+        996000  1250000
+        792000  1175000
+        396000  1175000
+        198000	1175000
+        >;
+    fsl,soc-operating-points = <
+        /* ARM kHz      SOC uV */
+        996000	1250000
+        792000	1175000
+        396000	1175000
+        198000	1175000
+        >;
+    arm-supply = <&reg_arm>;
+    soc-supply = <&reg_soc>;
+};
+
+&gpc {
+    /* use ldo-bypass, u-boot will check it and configure */
+    fsl,ldo-bypass = <0>;
+    fsl,wdog-reset = <1>; /* watchdog select of reset source */
+};
+
+&uart1 {
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_uart1>;
+    status = "okay";
+};
+
+&flexcan1 {
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_flexcan1>;
+    status = "okay";
+};
+
+&flexcan2 {
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_flexcan2>;
+    status = "okay";
+};
+
+&usbotg1 {
+	dr_mode = "host";
+	status = "okay";
+};
+
+&usbotg2 {
+	dr_mode = "peripheral";
+	status = "okay";
+};
+
+&usdhc3 {
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_usdhc3>;
+    bus-width = <4>;
+    cd-gpios = <&gpio7 6 GPIO_ACTIVE_HIGH>;
+    wp-gpios = <&gpio7 7 GPIO_ACTIVE_HIGH>;
+    status = "okay";
+};
+
+&gpmi {
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_gpmi_nand_1>;
+    status = "okay";
+    nand-on-flash-bbt;
+};
+
+&iomuxc {
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_hog>;
+
+    imx6x-score {
+pinctrl_hog: hoggrp {
+                 fsl,pins = <
+                     /* CAN1_EN */
+                     MX6SX_PAD_KEY_ROW4__GPIO2_IO_19     0xc9
+                     /* CAN2_EN */
+                     MX6SX_PAD_KEY_COL4__GPIO2_IO_14     0xc9
+                     /* CAN1_TERM */
+                     MX6SX_PAD_LCD1_DATA21__GPIO3_IO_22  0xc9
+                     /* CAN2_TERM */
+                     MX6SX_PAD_GPIO1_IO08__GPIO1_IO_8    0xc9
+                     /* 5P_EN */
+                     MX6SX_PAD_LCD1_DATA17__GPIO3_IO_18  0xc9
+                     >;
+             };
+
+pinctrl_gpmi_nand_1: gpmi-nand-1 {
+                         fsl,pins = <
+                             MX6SX_PAD_NAND_CLE__RAWNAND_CLE         0xb0b1
+                             MX6SX_PAD_NAND_ALE__RAWNAND_ALE         0xb0b1
+                             MX6SX_PAD_NAND_WP_B__RAWNAND_WP_B       0xb0b1
+                             MX6SX_PAD_NAND_READY_B__RAWNAND_READY_B 0xb000
+                             MX6SX_PAD_NAND_CE0_B__RAWNAND_CE0_B     0xb0b1
+                             MX6SX_PAD_NAND_RE_B__RAWNAND_RE_B       0xb0b1
+                             MX6SX_PAD_NAND_WE_B__RAWNAND_WE_B       0xb0b1
+                             MX6SX_PAD_NAND_DATA00__RAWNAND_DATA00   0xb0b1
+                             MX6SX_PAD_NAND_DATA01__RAWNAND_DATA01   0xb0b1
+                             MX6SX_PAD_NAND_DATA02__RAWNAND_DATA02   0xb0b1
+                             MX6SX_PAD_NAND_DATA03__RAWNAND_DATA03   0xb0b1
+                             MX6SX_PAD_NAND_DATA04__RAWNAND_DATA04   0xb0b1
+                             MX6SX_PAD_NAND_DATA05__RAWNAND_DATA05   0xb0b1
+                             MX6SX_PAD_NAND_DATA06__RAWNAND_DATA06   0xb0b1
+                             MX6SX_PAD_NAND_DATA07__RAWNAND_DATA07   0xb0b1
+                             >;
+                     };
+
+pinctrl_flexcan1: flexcan1grp {
+                      fsl,pins = <
+                          MX6SX_PAD_KEY_ROW2__CAN1_RX			0x1b020
+                          MX6SX_PAD_KEY_COL2__CAN1_TX			0x1b020
+                          >;
+                  };
+
+
+pinctrl_flexcan2: flexcan2grp {
+                      fsl,pins = <
+                          MX6SX_PAD_KEY_ROW3__CAN2_RX			0x1b020
+                          MX6SX_PAD_KEY_COL3__CAN2_TX			0x1b020
+                          >;
+                  };
+
+
+pinctrl_i2c1: i2c1grp {
+                  fsl,pins = <
+                      MX6SX_PAD_GPIO1_IO01__I2C1_SDA 0x4001b8b1
+                      MX6SX_PAD_GPIO1_IO00__I2C1_SCL 0x4001b8b1
+                      >;
+              };
+
+pinctrl_uart1: uart1grp {
+                   fsl,pins = <
+                       MX6SX_PAD_GPIO1_IO04__UART1_TX		0x1b0b1
+                       MX6SX_PAD_GPIO1_IO05__UART1_RX		0x1b0b1
+                       >;
+               };
+
+pinctrl_usdhc3: usdhc3grp {
+                    fsl,pins = <
+                        MX6SX_PAD_SD3_CMD__USDHC3_CMD		0x17069
+                        MX6SX_PAD_SD3_CLK__USDHC3_CLK		0x10071
+                        MX6SX_PAD_SD3_DATA0__USDHC3_DATA0	0x17069
+                        MX6SX_PAD_SD3_DATA1__USDHC3_DATA1	0x17069
+                        MX6SX_PAD_SD3_DATA2__USDHC3_DATA2	0x17069
+                        MX6SX_PAD_SD3_DATA3__USDHC3_DATA3	0x17069
+                        MX6SX_PAD_SD3_DATA4__GPIO7_IO_6		0x17059 /* CD */
+                        MX6SX_PAD_SD3_DATA5__GPIO7_IO_7		0x17059 /* WP */
+                        >;
+                };
+    };
+};
-- 
1.9.1

